x/tx/signing/aminojson/encoder.go

```diff
@@ -51,7 +53,12 @@ func cosmosDecEncoder(_ *Encoder, v protoreflect.Value, w io.Writer) error {
 		if val == "" {
 			return jsonMarshal(w, "0")
 		}
-		return jsonMarshal(w, val)
+		var dec math.LegacyDec
+		err := dec.Unmarshal([]byte(val))
```

Comment:
Dynamic messages represent the packed Decimal value without a radix, so `math` marshaling (`String()` implementation) is needed here.

Reply:
Great, I can actually delete the custom encoder here then: https://github.com/cosmos/cosmos-sdk/pull/21853 hehe